acestep/text_tasks/external_lm_model_discovery.py

```
@@ -0,0 +1,151 @@
+        base_url=base_url,
+    )
+    if not candidate_urls:
+        raise ExternalModelDiscoveryError(
+            "No model-discovery endpoint could be derived from base URL."
+        )
+
+    headers = _build_auth_headers(protocol=protocol, api_key=api_key)
+    failures: list[str] = []
+
+    for url in candidate_urls:
+        req = request.Request(url=url, headers=headers, method="GET")
```

Ruff (0.15.6)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)